.github/workflows/ensure_linting.yml

```diff
@@ -0,0 +1,25 @@
+name: Ensure Linting
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  check_linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b #v27
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Nix binary cache
+        uses: DeterminateSystems/magic-nix-cache-action@87b14cf437d03d37989d87f0fa5ce4f5dc1a330b #v8
+
+      - name: Run treefmt in CI mode
+        run: nix develop --override-input devenv-root "file+file://"<(printf %s "$PWD") . --command treefmt --ci
```

Comment:
I see we're using `nix develop`  to run treefmt, we could probably just do `nix build .#checks.x86_64-linux.treefmt` instead which should be a more reproducible result since then `treefmt` will run inside the nix sandbox.

Reply:
Yes and no. `mix format` does need the dependencies fetched beforehand. Running `nix build .#checks.aarch64-darwin.treefmt` locally results therefore in:

```bash
❯ nix build .#checks.aarch64-darwin.treefmt
warning: Git tree '/Users/jakob/development/repos/teslamate' is dirty
error: builder for '/nix/store/3nj3634djh1s9acq79jd04n17ki0v2w2-treefmt-check.drv' failed with exit code 1;
       last 7 log lines:
       > Leeres Git-Repository in /private/tmp/nix-build-treefmt-check.drv-0/project/.git/ initialisiert
       > treefmt v2.0.5
       > mix-format error:
       > Mix requires the Hex package manager to fetch dependencies
       > Shall I install Hex? (if running non-interactively, use "mix local.hex --force") [Yn] ** (Mix) Could not find an SCM for dependency :castore from TeslaMate.MixProject
       >
       > treefmt: error: formatting failure: formatter '/nix/store/irh1cr1a71355gl7z329qx5fp3kmqjbr-elixir-1.17.3/bin/mix' with options '[format]' failed to apply: exit status 1
       For full logs, run 'nix log /nix/store/3nj3634djh1s9acq79jd04n17ki0v2w2-treefmt-check.drv'.
```

That's why I make use of the devshell to first pull the dependencies.

Reply:
Oh that's lame... There's probably a way to make it work correctly with the check, but I doubt it's particularly simple.... 

In that case, we should probably disable the check (by setting `treefmt.flakeCheck` to false). IMO having a check that is known to not work is worse than having no check at all.

Reply:
I think I can imagine a possible solution here. But lets get this PR merged first.

We probably should set `treefmt.flakeCheck = false` for now though. Marking as unresolved so this doesn't get missed.

Reply:
Agree and  I have implemented it.